openam-shared/src/main/java/com/sun/identity/shared/debug/Debug.java

```diff
@@ -259,38 +261,45 @@ static IDebugProvider getDebugProvider() {
      */
     private static synchronized void initialize() {
         if (!serviceInitialized) {
-            String providerName = SystemPropertiesManager.get(DebugConstants.CONFIG_DEBUG_PROVIDER);
+            final String providerName;
             IDebugProvider provider = null;
             boolean providerLoadFailed = false;
-            Exception exceptionCatched = null;
-            if (providerName != null && providerName.trim().length() > 0) {
+            Exception exceptionCaught = null;
+
+            providerName =
+                Optional.ofNullable(
+                    SystemPropertiesManager.get(DebugConstants.CONFIG_DEBUG_PROVIDER)
+                ).map(String::trim)
+                .orElse(null);
+
+            if (!Strings.isNullOrEmpty(providerName)) {
                 try {
-                    provider = (IDebugProvider) Class.forName(providerName).newInstance();
-                } catch (ClassNotFoundException cnex) {
+                    provider = (IDebugProvider)Class.forName(providerName).newInstance();
```

Comment:
I realize that this code was already here but isn't `newInstance` considered evil? The [docs](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#newInstance--) say:

> Note that this method propagates any exception thrown by the nullary constructor, including a checked exception. Use of this method effectively bypasses the compile-time exception checking that would otherwise be performed by the compiler. The Constructor.newInstance method avoids this problem by wrapping any exception thrown by the constructor in a (checked) InvocationTargetException.